src/server/game/Spells/Spell.cpp

```diff
@@ -3001,7 +3001,9 @@ void Spell::prepare(SpellCastTargets const* targets, AuraEffect const* triggered
     if (m_caster->GetTypeId() == TypeID::TYPEID_PLAYER)
         m_caster->ToPlayer()->SetSpellModTakingSpell(this, true);
     // Fill cost data (not use power for item casts)
-    m_powerCost = m_CastItem ? 0 : m_spellInfo->CalcPowerCost(m_caster, m_spellSchoolMask);
+    if (!m_powerCost || (m_spellInfo->ShapeShiftReqID && !m_caster->ToPlayer()->GetAura(m_spellInfo->ShapeShiftReqID)))
```

Comment:
m_powerCost is allways empty, so !m_powerCost will allways return true, because we just setting that value below

Reply:
the point of the shapeshift requirement check is to not use the m_spellInfo data if the caster does not have the required shapeshift aura

Reply:
https://www.wowhead.com/spell=5215/prowl

if what you wrote would work then this spell would work and that is, we would go into the form when using it?

Reply:
I will look in another source and redo the PR

Reply:
we shouldn't forcefully establish !m_powerCost=0?

Reply:
m_powerCost is a protected member variable from Spell and is initialized as 0 by default. in the code here we setting the variable for the first time. so your if (!m_powerCost) is a hack to forcefully call the CalcValue function from spellInfo